tokio/src/signal/unix.rs

```diff
@@ -23,13 +23,25 @@ pub(crate) type OsStorage = Box<[SignalInfo]>;
 impl Init for OsStorage {
     fn init() -> Self {
         // There are reliable signals ranging from 1 to 33 available on every Unix platform.
-        #[cfg(not(target_os = "linux"))]
+        #[cfg(not(any(target_os = "linux", target_os = "illumos")))]
         let possible = 0..=33;
 
         // On Linux, there are additional real-time signals available.
         #[cfg(target_os = "linux")]
         let possible = 0..=libc::SIGRTMAX();
 
+        // On illumos, signal numbers go up to 41 (SIGINFO). The list of signals
+        // hasn't changed since 2013. See
+        // https://github.com/illumos/illumos-gate/blob/master/usr/src/uts/common/sys/iso/signal_iso.h.
+        //
+        // illumos also has real-time signals, but (a) the number of real-time
+        // signals is actually configurable at runtime and (b) this capability
+        // isn't exposed by libc as of 0.2.167, so we don't support them at the
+        // moment. If support for real-time signals on illumos is desired, this
+        // code would have to be changed to accommodate that.
```

Comment:
Thanks for including the comment, this is lovely!

Out of interest, there wouldn't happen to be any libc ticket or similar tracking support for real-time issues? Not a big deal, but if there is something, it could be worth referencing that?

Reply:
No, I couldn't find a libc issue.

Reply:
Put up https://github.com/rust-lang/libc/pull/4171.

Reply:
Added a link -- once it lands in upstream libc, it should be possible to do on illumos what's done on Linux.